### PR TITLE
Ros 325 wrapper optimized

### PR DESCRIPTION
**Pull request by VHolonec**
Added another implementation using libiio api.

### CODE REVIEW

File: src/iio_wrapper.cpp [hunk 1]
```diff
@@ -6,76 +6,81 @@
...
+    m_channel_anglvel_x = iio_device_find_channel(m_dev, "anglvel_x", false);
+    m_channel_anglvel_y = iio_device_find_channel(m_dev, "anglvel_y", false);
+    m_channel_anglvel_z = iio_device_find_channel(m_dev, "anglvel_z", false);
 
 }
 
-std::string IIOWrapper::ssystem(const char *command,const char *filename)
+IIOWrapper::~IIOWrapper()
 {
+    iio_context_destroy(m_object_context);
```
Comment (dNechita) [inline src/iio_wrapper.cpp:35]: you also need to destroy the m_network_context. And also to make sure other instances of IIOWrapper will not attempt to destroy it as well.

File: src/iio_wrapper.cpp [hunk 2]
```diff
@@ -12,7 +12,12 @@ IIOWrapper::IIOWrapper()
 {
     if(m_network_context == nullptr)
     {
+        #ifdef __arm__
```
Comment (dNechita) [inline src/iio_wrapper.cpp]: Please remove the #if-def directives. Just use the this: iio_create_context_from_uri("ip:127.0.0.1");
Once you have an implementation that takes an URI from command line, you can replace "ip:127.0.0.1" with the provided URI.

**Comment by mhennerich** [comment]
likewise - if you want local use do iio_create_context_from_uri("local:")

			Use the context at the provided URI.
			eg: 'ip:192.168.2.1', 'ip:pluto.local', or 'ip:'
			    'usb:1.2.3', or 'usb:'
			    'serial:/dev/ttyUSB0,115200,8n1'
			    'local:' (Linux only)

File: src/iio_wrapper.cpp (hunk 2 above)
Comment (VHolonec) [reply src/iio_wrapper.cpp]: I changed the directive wiht uri in another commit.

**Review by dNechita — APPROVED** [review]

**Pull request merged**